### DISCUSSION TITLE
Having trouble with useContact / useContractWrite

### DISCUSSION THREAD

**Discussion by EzraWeller** [wevm/wagmi#49] [Q&A]
Hey all,

So much <3 for making this library. Been trying to integrate it with a project today and I'm coming up short trying to instantiate a contract connected to a signer. 

Attempting to copy the docs / example, I've been trying:
```
(contractAddress: string, abi: ContractInterface, network: string): Contract | string => {
  const contract: Contract = useContract({
    addressOrName: contractAddress,
    contractInterface: abi
  })
  const provider = useProvider()
  if (noConnection(provider)) return WARNINGS.no_connection
  if (wrongNetwork(provider, network)) return WARNINGS.wrong_network
  return contract
}
```
But I only seem to get `signer: null` in the result. I see I can get the signer asynchronously from the result of `useAccount` and then plug that in to `useContract`, but that doesn't seem like the intended way to do this.

Just tried `useContractWrite` in the same way and got the same result.

What am I missing?

**Comment by clifton** [comment]
i think this is one area where the wagmi api falls down vs `contract = useContract<EnsContract>(); await contract.doSomething()`

**Reply by tmm** [reply]
What do you mean?

**Comment by jasheal** [comment]
The hook provides the write method to call as the second arg. The first arg has data after the user has signed. The loading shows after the write method is called and before signing tx. 

```
const [{ data: sendWrapTransactionData, loading: waitForSignature }, write] =
    useContractWrite(
      {
        addressOrName: '0xb603cEa165119701B58D56d10D2060fBFB3efad8',
        contractInterface: weth9ABI,
      },
      'deposit'
    )

const wrap = () => {
    write({ args: { value: utils.parseEther(wrapAmount) } })
  } 

useEffect(() => {
    if (sendWrapTransactionData) {
      viewScreenById(SCREEN_IDS.WRAP_ETH_LOADING, {
        hash: sendWrapTransactionData.hash,
        amount: wrapAmount,
      })
    }
  }, [sendWrapTransactionData, viewScreenById, wrapAmount])

```
